packages/node_modules/pouchdb-core/src/adapter.js

```diff
@@ -78,13 +74,12 @@ function cleanDocs(docs) {
 
 // compare two docs, first by _id then by _rev
 function compareByIdThenRev(a, b) {
-  var idCompare = compare(a._id, b._id);
-  if (idCompare !== 0) {
-    return idCompare;
+  if (a._id === b._id) {
+   const aStart = a._revisions ? a._revisions.start : 0;
+   const bStart = b._revisions ? b._revisions.start : 0;
+   return aStart - bStart;
```

Comment:
The indentation is off I think...

```suggestion
    const aStart = a._revisions ? a._revisions.start : 0;
    const bStart = b._revisions ? b._revisions.start : 0;
    return aStart - bStart;
```